flang/lib/Lower/OpenMP.cpp

```diff
@@ -88,29 +116,64 @@ genOMP(Fortran::lower::AbstractConverter &absConv,
     auto &firOpBuilder = absConv.getFirOpBuilder();
     auto currentLocation = absConv.getCurrentLocation();
 
-    // Clauses.
-    // FIXME: Add support for other clauses.
-    mlir::Value numThreads;
+    mlir::Value ifClauseOperand, numThreadsClauseOperand;
+    SmallVector<Value, 4> privateClauseOperands, firstprivateClauseOperands,
+        sharedClauseOperands, copyinClauseOperands;
+    Attribute defaultClauseOperand, procBindClauseOperand;
 
     const auto &parallelOpClauseList =
         std::get<Fortran::parser::OmpClauseList>(blockDirective.t);
     for (const auto &clause : parallelOpClauseList.v) {
-      if (const auto &numThreadsClause =
-              std::get_if<Fortran::parser::OmpClause::NumThreads>(&clause.u)) {
+      if (const auto &ifClause =
+              std::get_if<Fortran::parser::OmpIfClause>(&clause.u)) {
+        auto &expr = std::get<Fortran::parser::ScalarLogicalExpr>(ifClause->t);
+        ifClauseOperand = fir::getBase(
+            absConv.genExprValue(*Fortran::semantics::GetExpr(expr)));
+      } else if (const auto &numThreadsClause =
+                     std::get_if<Fortran::parser::OmpClause::NumThreads>(
+                         &clause.u)) {
         // OMPIRBuilder expects `NUM_THREAD` clause as a `Value`.
-        numThreads = fir::getBase(absConv.genExprValue(
+        numThreadsClauseOperand = fir::getBase(absConv.genExprValue(
             *Fortran::semantics::GetExpr(numThreadsClause->v)));
+      } else if (const auto &defaultClause =
+                     std::get_if<Fortran::parser::OmpDefaultClause>(
+                         &clause.u)) {
+        TODO();
```

Comment:
We're moving to `TODO("string")` in lowering. The string can be anything, including "", of course.